### PR TITLE
[Merged by Bors] - feat(NumberTheory/PrimeCounting): Lemma: The `n`th prime is greater or equal `n + 2`.

### PR DESCRIPTION
**Pull request by rwst**
Possibly useful lemma about the nth prime number.

Co-authored-by: Pol'tta / Miyahara Kō <52843868+Komyyy@users.noreply.github.com>
Co-authored-by: Thomas Browning <tb65536@uw.edu>

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/NumberTheory/PrimeCounting.lean [hunk 1]
```diff
@@ -105,4 +105,11 @@ theorem primeCounting'_add_le {a k : ℕ} (h0 : 0 < a) (h1 : a < k) (n : ℕ) :
 @[simp]
 theorem zeroth_prime_eq_two : nth Prime 0 = 2 := nth_count prime_two
 
+/-- The `n`th prime is greater or equal `n + 2`. -/
+lemma nth_prime_ge_add_two (n : ℕ): nth Prime n ≥ n + 2 := by
+  induction' n with n ih
+  · exact le_iff_eq_or_lt.mpr (Or.inl zeroth_prime_eq_two.symm)
+  · exact LE.le.trans (ge_iff_le.mp ((add_le_add_iff_right 1).mpr ih))
+      (succ_le_iff.mpr <| (nth_lt_nth infinite_setOf_prime).mpr <| lt_add_one n)
```
Comment (Komyyy) [inline Mathlib/NumberTheory/PrimeCounting.lean:113]:
```suggestion
/-- The `n`th prime is greater or equal `n + 2`. -/
lemma add_two_le_nth_prime (n : ℕ) : n + 2 ≤ nth Prime n := by
  induction' n with n ih
  · exact ge_of_eq zeroth_prime_eq_two
  · exact le_trans (add_le_add_right ih 1)
      (succ_le_of_lt <| nth_strictMono infinite_setOf_prime n.lt_succ_self)
```
`GT.gt`(`a ≥ b`) is defined as an abbreviation of `LT.lt` (`b ≤ a`) so using the latter is more preferable.
Also, we can golf (working piece of a proof as short as possible) here.

**Review by Komyyy — CHANGES_REQUESTED** [review]

**Comment by Komyyy** [comment]
I misunderstood that a non-reviewer could ask a maintainer to merge with the command “maintainer merge”.
I am not a reviewer and apologize for the review even though I do not have the authority to do the merge.

**Comment by rwst** [comment]
Your review is welcome! I'm surprised how much you could shorten the proof. Thanks!

**Comment by erdOne** [comment]
@Komyyy Speaking as a member of the reviewer team: we encourage and are grateful for every reviews! It makes our work easier and also let PR authors wait less. If you think this PR (or any other one) is good to go, give it an "approve" and a reviewer / maintainer will get to it soon.

**Comment by erdOne** [comment]
I wonder if we should have `n + f 0 <= f n` if `f : ℕ → ℕ` is `StrictMono`.

**Review by Komyyy — APPROVED** [review]

**Comment by tb65536** [comment]
> I wonder if we should have `n + f 0 <= f n` if `f : ℕ → ℕ` is `StrictMono`.

Or slightly more generally:
```
lemma StrictMono.add_le {f : ℕ → ℕ} (hf : StrictMono f) (m n : ℕ) : m + f n ≤ f (m + n)  := by
  rw [add_comm m, add_comm m]
  induction' m with m ih
  · rw [add_zero, add_zero]
  · rw [← add_assoc, ← add_assoc, succ_le]
    exact ih.trans_lt (hf (n + m).lt_succ_self)

/-- The `n`th prime is greater or equal to `n + 2`. -/
lemma add_two_le_nth_prime (n : ℕ) : n + 2 ≤ nth Prime n :=
  zeroth_prime_eq_two ▸ (nth_strictMono infinite_setOf_prime).add_le n 0
```

File: Mathlib/Order/Monotone/Basic.lean [hunk 2]
```diff
@@ -626,6 +626,13 @@ theorem StrictAnti.isMin_of_apply (hf : StrictAnti f) (ha : IsMax (f a)) : IsMin
     (hf hb).not_isMax ha
 #align strict_anti.is_min_of_apply StrictAnti.isMin_of_apply
 
+lemma StrictMono.add_le {f : ℕ → ℕ} (hf : StrictMono f) (m n : ℕ) : m + f n ≤ f (m + n)  := by
```
Comment (urkud) [inline Mathlib/Order/Monotone/Basic.lean:629]: This is true for some other types as well (e.g., for `Int`, if you assume `0 ≤ m`), so I suggest that you add "_nat" to the name of this lemma.

Comment (rwst) [reply Mathlib/Order/Monotone/Basic.lean]: Done.

**Comment by rwst** [comment]
Apparently the name change in the last commit clashes with the PR comment? What to do?

**Comment by rwst** [comment]
OK, so the fail is ignored. Let's see...

**Comment by MichaelStollBayreuth** [comment]
LGTM.
maintainer merge

**Comment by kbuzzard** [comment]
Thanks!

bors merge

**Comment by erdOne** [comment]
Wait what?

**Pull request merged**